cmd/thv/app/run.go

```diff
@@ -245,6 +245,11 @@ func runCmdFunc(cmd *cobra.Command, args []string) error {
 		return fmt.Errorf("failed to retrieve or pull image: %v", err)
 	}
 
+	// pull the egress image if it is not already pulled
```

Comment:
This needs to be moved inside the lifecycle manager, but that can wait till a future PR since more refactoring is needed in this part of the code.